src/Riok.Mapperly/Symbols/SetterMemberPath.cs

```diff
@@ -0,0 +1,78 @@
+﻿using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Riok.Mapperly.Descriptors;
+
+namespace Riok.Mapperly.Symbols;
+
+public class SetterMemberPath : MemberPath
+{
+    private SetterMemberPath(IReadOnlyList<IMappableMember> path, bool isMethod)
+        : base(path)
+    {
+        IsMethod = isMethod;
+    }
+
+    public bool IsMethod { get; }
+
+    public static SetterMemberPath Build(MappingBuilderContext ctx, MemberPath memberPath)
+    {
+        var path = memberPath.Path;
+        var array = new IMappableMember[path.Count];
+
+        var isMethod = false;
+        for (var i = 0; i < path.Count; i++)
+        {
+            var item = path[i]!;
+            if (ctx.SymbolAccessor.IsAccessible(item.MemberSymbol))
+            {
+                array[i] = item;
+                continue;
+            }
+
+            if (item.MemberSymbol.Kind == SymbolKind.Field)
+            {
+                var unsafeAccessor = ctx.UnsafeAccessorContext.BuildGetField(
+                    item.MemberSymbol.ContainingType,
+                    (IFieldSymbol)item.MemberSymbol
+                );
+
+                array[i] = new MethodMember(item, unsafeAccessor.MethodName);
+            }
+            else if (i < path.Count - 1)
+            {
+                var unsafeAccessor = ctx.UnsafeAccessorContext.BuildGetProperty(
+                    item.MemberSymbol.ContainingType,
+                    (IPropertySymbol)item.MemberSymbol
+                );
+
+                array[i] = new MethodMember(item, unsafeAccessor.MethodName);
+            }
+            else
+            {
+                isMethod = true;
+                var unsafeAccessor = ctx.UnsafeAccessorContext.BuildSetProperty(
+                    item.MemberSymbol.ContainingType,
+                    (IPropertySymbol)item.MemberSymbol
+                );
+
+                array[i] = new MethodMember(item, unsafeAccessor.MethodName, isMethod: true);
+            }
+        }
+
+        return new SetterMemberPath(array, isMethod);
+    }
+
+    public ExpressionSyntax BuildAccess(ExpressionSyntax? baseAccess)
+    {
+        IEnumerable<IMappableMember> path = Path;
+
+        if (baseAccess == null)
+        {
+            baseAccess = SyntaxFactory.IdentifierName(path.First().Name);
```

Comment:
Not possible `path` is an `IEnumerable` type. It can't be `IReadOnlyList` because this prevents `Skip` from working. 
There are probably work arounds if you want me to explore them, although the original `MemberPath` code never used them

Reply:
The variable path is reassigned to path.Skip(1) if baseAccess is null. It has to be a local variable and must be IEnumerable<IMappableMember> for the skip to work.

I'm pretty sure I copied this code straight from the original memberPath